### PR TITLE
get timestamp for versioncode from last git commmit

### PR DESCRIPTION
**Pull request by Bubu**
This makes the versioncode for an apk reproducible. Building the same
commit/code will produce an apk with the same VC.

This is a requirement for being able to build pre-release version by f-droid.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -15,9 +15,18 @@ def versionMajor = 0
 def versionMinor = 1
 def versionPatch = 0
 
+def getGitTimetamp() {
```
Comment (bmarty) [inline app/build.gradle]: A `s`is missing :)
Also I prefer this format, more readable: https://github.com/vector-im/riot-android/blob/develop/vector/build.gradle#L193

File: app/build.gradle [hunk 2]
```diff
@@ -15,9 +15,18 @@ def versionMajor = 0
...
+    def stdout = new ByteArrayOutputStream()
+    exec {
+        commandLine 'git', 'show', '-s', '--format=%ct'
+        standardOutput = stdout
+    }
+    return stdout.toString() as Integer
```
Comment (bmarty) [inline app/build.gradle]: Timestamp should be `Long`

File: app/build.gradle [hunk 3]
```diff
@@ -15,9 +15,18 @@ def versionMajor = 0
...
+}
+
 def generateVersionCodeFromTimestamp() {
     // It's unix timestamp divided by 10: It's incremented by one every 10 seconds.
```
Comment (bmarty) [inline app/build.gradle]: This comment should be updated too

File: app/build.gradle (hunk 1 above)
Comment (Bubu) [reply app/build.gradle]: > A sis missing :)

:see_no_evil: 

>  Also I prefer this format, more readable: https://github.com/vector-im/riot-android/blob/develop/vector/build.gradle#L193

What exactly do you mean here? The syntax for the gradle function? I can change that, sure.

File: app/build.gradle (hunk 3 above)
Comment (Bubu) [reply app/build.gradle]: I deliberately left the comment in place, I think it's still valid. How would you change it?

File: app/build.gradle (hunk 1 above)
Comment (bmarty) [reply app/build.gradle]: Yes, it was the format of the gradle function, thanks for the change

File: app/build.gradle (hunk 3 above)
Comment (bmarty) [reply app/build.gradle]: The comment can stay like that. The point is that the value doesn't change anymore if there is no new commit.

**Comment by bmarty** [comment]
Thanks for the update, all I need now to merge this is a Sign-Off please.

**Comment by Bubu** [comment]
This should do it?

**Comment by bmarty** [comment]
Yes, thanks, I will merge the PR soon

**Pull request merged**